stablehlo/reference/Element.cpp

```diff
@@ -334,44 +338,132 @@ Element Element::operator/(const Element &other) const {
                                      debugString(type).c_str()));
 }
 
-Element Element::operator*(const Element &other) const {
-  return map(
-      *this, other, [](APInt lhs, APInt rhs) { return lhs * rhs; },
-      [](bool lhs, bool rhs) -> bool { return lhs & rhs; },
-      [](APFloat lhs, APFloat rhs) { return lhs * rhs; },
-      [](std::complex<APFloat> lhs, std::complex<APFloat> rhs) {
-        // TODO(#226): Use std::complex::operator*
-        auto resultReal = lhs.real() * rhs.real() - lhs.imag() * rhs.imag();
-        auto resultImag = lhs.real() * rhs.imag() + lhs.imag() * rhs.real();
-        return std::complex<APFloat>(resultReal, resultImag);
-      });
+bool Element::operator<(const Element &other) const {
+  Type type = other.getType();
+  if (type_ != type)
+    report_fatal_error(invalidArgument("Element types don't match: %s vs %s",
+                                       debugString(type_).c_str(),
+                                       debugString(type).c_str()));
+
+  if (isSupportedIntegerType(type)) {
+    auto intLhs = getIntegerValue();
+    auto intRhs = other.getIntegerValue();
+    return isSupportedSignedIntegerType(type) ? intLhs.slt(intRhs)
+                                              : intLhs.ult(intRhs);
+  }
+
+  if (isSupportedBooleanType(type)) {
+    auto boolLhs = getBooleanValue();
+    auto boolRhs = other.getBooleanValue();
+    return boolLhs < boolRhs;
+  }
+
+  if (isSupportedFloatType(type)) {
+    auto floatLhs = getFloatValue();
+    auto floatRhs = other.getFloatValue();
+    return floatLhs < floatRhs;
+  }
+
+  if (isSupportedComplexType(type)) {
+    auto complexLhs = getComplexValue();
+    auto complexRhs = other.getComplexValue();
+    if (complexLhs.real().isNaN() || complexLhs.imag().isNaN() ||
```

Comment:
These checks of `NaN` values are redundant. The logic below should help returning the desired false?

Reply:
same for `operator>`